plugins/woocommerce/templates/single-product/add-to-cart/grouped.php

```diff
@@ -64,7 +64,14 @@
 								woocommerce_template_loop_add_to_cart();
 							} elseif ( $grouped_product_child->is_sold_individually() ) {
 								echo '<input type="checkbox" name="' . esc_attr( 'quantity[' . $grouped_product_child->get_id() . ']' ) . '" value="1" class="wc-grouped-product-add-to-cart-checkbox" id="' . esc_attr( 'quantity-' . $grouped_product_child->get_id() ) . '" />';
-								echo '<label for="' . esc_attr( 'quantity-' . $grouped_product_child->get_id() ) . '" class="screen-reader-text">' . esc_html__( 'Buy one of this item', 'woocommerce' ) . '</label>';
+								echo '<label for="' . esc_attr( 'quantity-' . $grouped_product_child->get_id() ) . '" class="screen-reader-text">' . 
+									sprintf(
+										esc_html__( 'Buy one of %1$s for %2$s%3$s', 'woocommerce' ),
+										$grouped_product_child->get_name(),
+										get_woocommerce_currency_symbol(),
+										$grouped_product_child->get_price()
```

Comment:
This combination works only for some currencies. What about using `$grouped_product_child->get_price_html()` which does all the stuff to format the price properly and then running `wp_strip_all_tags` on that?

Reply:
Thanks for the tip!